ggshield/iac/output/iac_text_output_handler.py

```diff
@@ -1,25 +1,58 @@
 import shutil
+from collections import defaultdict
 from io import StringIO
 from pathlib import Path
-from typing import ClassVar, List
+from typing import ClassVar, Dict, Generator, List, NamedTuple, Optional
 
 from pygitguardian.iac_models import IaCFileResult, IaCVulnerability
 
 from ggshield.core.text_utils import (
     STYLE,
     Line,
     clip_long_line,
+    file_diff_info,
     file_info,
     format_text,
     get_offset,
     get_padding,
+    pluralize,
 )
 from ggshield.core.utils import Filemode, get_lines_from_content
 from ggshield.iac.collection.iac_diff_scan_collection import IaCDiffScanCollection
 from ggshield.iac.collection.iac_path_scan_collection import IaCPathScanCollection
+from ggshield.iac.iac_scan_models import IaCDiffScanEntities
+from ggshield.iac.output.iac_output_handler import IaCOutputHandler
 from ggshield.scan import File
 
-from .iac_output_handler import IaCOutputHandler
+
+class GroupedIncidents(NamedTuple):
+    name: str
+    new: List[IaCFileResult]
+    unchanged: List[IaCFileResult]
+    deleted: List[IaCFileResult]
+
+
+def group_incidents_by_filename(
+    incidents: IaCDiffScanEntities,
+) -> Generator[GroupedIncidents, None, None]:
+    filenames = []
+    statuses: Dict[str, dict] = {
```

Comment:
incorrect typing, should use `defaultdict` instead of `dict`.
And you shouldn't use `dict` for typing as this is not supposed to be compatible with python3.7 (still supported by ggshield)